text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,799 @@
+- Allow semantic convention resource modeling to progress ([spec#605](https://github.com/open-telemetry/opentelemetry-specification/issues/605), [spec#559](https://github.com/open-telemetry/opentelemetry-specification/issues/559), etc).
+
+## Design
+
+### Approach - Resource Improvements
+
+Let's focus on outlining Entity detectors and Resource composition.  This has a higher priority for fixing within OpenTelemetry, and needs to be unblocked sooner. Then infer our way back to data model and Collector use cases.
+
+We define the following SDK components:
+
+- **Resource Detectors (legacy)**:  We preserve existing resource detectors.  They have the same behavior and interfaces as today.
+- **Entity Detectors (new)**: Detecting an entity that is relevant to the current instance of the SDK.  For example, this would detect a service entity for the current SDK, or its process. Every entity must have some relation to the current SDK.
+- **Resource Coordinator (new)**: A component responsible for taking Resource and Entity detectors and doing the following:
+  - Constructing a Resource for the SDK from detectors.
+  - Dealing with conflicts between detectors.
+  - Providing SDK-internal access to detected Resources for reporting via Log signal on configured LogProviders.
+  - *(new) Managing Entity changes during SDK lifetime, specifically dealing with entities that have lifetimes shorter than the SDK*
+
+#### Resource Coordinator
+
+The SDK Resource Coordinator is responsible for running all configured Resource and Entity Detectors.  There will be some (user-controlled, otel default) priority order to these.
+
+- The Resource Coordinator will detect conflicts in Entity of the same type being discovered and choose one to use.
+- When using Entity Detectors and Resource detectors together, the following merge rules will be used:
+  - Entity merging will occur first resulting in an "Entity Merged" Resource (See [algorithm here](#entity-merging-and-resource)).
+  - Resource detectors otherwise follow existing merge semantics.
+    - The Specification merge rules will be updated to account for violations prevalent in ALL implementation of resource detection.
+    - Specifically: This means the [rules around merging Resource across schema-url will be dropped](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#merge).  Instead only conflicting attributes will be dropped.
+    - SchemaURL on Resource will need to be deprecated with entity-specific schema-url replacing it. SDKs will no longer fill out SchemaURL on Resource. Additionally, as no (non-service) Resource semantic conventions have ever stabilized, SchemaURL usage on Resource cannot be in stable components of OpenTelemetry.  Given prevalent concerns of implementations around Resource merge specification, we suspect impact of this deprecation to be minimal.
```

Comment:
I think SchemaURL field is part of Stable specification document: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md, which means this is a breaking change, I don't think we are allowed that.

Is there a way we can continue including SchemaURL on the Resource? It does not need to perfect (it is not perfect today). For example, I think we can do this: 

If all Entities use the same SchemaURL we use that value for Resource's SchemaURL too. If there are any difference in SchemaURL values of Entities than we leave Resource's SchemaURL value empty. This will be no worse than the currently specified behavior of Resource merging where attempting to merge with different SchemaURLs is [considered an error](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#merge).

I think this behavior is backward compatible and we are not breaking the "Stable" document promise.

Reply:
Wouldn't marking it for deprecation, but not removing it equally not break the "stable" document promise? It would cause it to remain until some v2.0.0 API is released, but the intention would be expressed clearly.

Reply:
I updated this description.

As  a note though, the ONLY resources to have stabilized are Service + Telemetry SDK, so any other usage of SchemaURL within open-telemetry is NOT stable, and therefore not a breaking change, according to the rules we've set forth in our documents.

Reply:
Marking this as resolved, please reopen if you see continued issues with how the SchemaURL section is phrased.